rasa/core/processor.py

```diff
@@ -193,6 +198,9 @@ async def get_tracker_with_session_start(
         if not tracker:
             return None
 
+        if metadata:
```

Comment:
I think it makes sense to put that into `_update_tracker_session `, after `if` statement on the line 169. Then it's very clear that we add metadata only for the first message that triggers the session.
What do you think?

Reply:
You're right!

This is my first PR on an open-source project, and this is my first implementation in Python, had a lot on my head so I missed this completely, sorry.